provider/marathon/marathon.go

```diff
@@ -537,3 +676,12 @@ func parseIndex(index string, length int) (int, error) {
 
 	return parsed, nil
 }
+
+func extractServiceName(varg ...string) string {
+	switch {
+	case len(varg) == 1:
+		return varg[0]
+	default:
```

Comment:
I think we've established before that template API stability is not a concern. As such, we can extend the existing signatures to serve our long-term maintainability needs.